balboa-client/src/main/java/com/blist/metrics/impl/queue/MetricsBuffer.java

```diff
@@ -0,0 +1,89 @@
+package com.blist.metrics.impl.queue;
+
+import com.socrata.balboa.metrics.Metrics;
+import com.socrata.metrics.MetricQueue$;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+
+/**
+ * An in-memory metrics buffer that is optimized to condense the metric message footprint.  This is
+ * current done by condensing the memory footprint of the metrics themselves.  Depending on the
+ * {@link com.socrata.balboa.metrics.Metric.RecordType}, the buffer manipulates the data to conserve space
+ * in memory.
+ *
+ * Note: Be very cautious using the Class.  Try to avoid it if you can.  Mutating metrics data in memory in the
+ * event of a killed process can result in lost, unrecoverable data.
+ *
+ * @see Metrics
+ * @see com.socrata.balboa.metrics.Metric
+ * @see com.socrata.balboa.metrics.Metric.RecordType
+ *
+ * Created by michaelhotan on 1/29/16.
+ */
+public class MetricsBuffer {
+    private static final Logger log = LoggerFactory.getLogger(MetricsBuffer.class);
+
+    /**
+     * Class initially copied from Metric Consumer in Coreserver to MetricJmsQueueNotSingleton to here.
+     *
+     * This class was created in an effort to prevent the overloading of a Queueing system.  For legacy
+     * reasons we continue to support it.   This class a couple of known risks that we look to mitigate in
+     * the future.
+     * * Class mutates metrics data in memory.
+     * * Additional buffering layer that in memory and can result to non recoverable metric loss.
```

Comment:
First answer, yes.  

Second answer, yes.  Metric event logs can be read from disk, transmitted, and process crashes with a non IOException and the consumer fails to mark the file as broken and the process restarts later.